cumulusci/core/flowrunner.py

```diff
@@ -628,33 +630,37 @@ def _check_old_yaml_format(self):
             else:
                 raise FlowConfigError("No steps found in the flow definition")
 
-    def _check_infinite_flows(self, flow_config, visited_flows=None):
+    def _check_infinite_flows(self, flow_config, flow_stack=None):
         """
         Recursively loop through the flow_config and check if there are any cycles.
 
-        :param steps: Set of step definitions to loop through
-        :param flows: Flows already visited.
+        :param flow_config: FlowConfig to traverse to find cycles/infinite loops
+        :param flow_stack: list of flow signatures already visited
         :return: None
         """
-        if visited_flows is None:
-            visited_flows = set()
+        if not flow_stack:
+            flow_stack = []
         project_config = flow_config.project_config
+
         for step in flow_config.steps.values():
             if "flow" in step:
                 flow_name = step["flow"]
                 if flow_name == "None":
                     continue
```

Comment:
This is odd. Running `make coverage` shows that we have 100% code coverage on this file (nice job!), but when I step through these two tests I don't see that this case is triggered. @davisagli - I'm guessing this has to do with flows listed as `None` in the config file to skip a step. Would it be worth it for @davidjray to add another flow that would trigger this case?

Reply:
It looks like this is covered by `SimplTestFlowCoordinator.test_init_ambiguous_step()`, so I'm guessing that you don't really need to explicitly test this case, but let's see what David thinks.

Reply:
that existing test looks sufficient